File: frontend/src/types/session.ts
export interface Session {
  id: number;
  course_id: number;
  course_name?: string;
  number: number;
  title: string;
  description?: string;
  created_at: string;
  updated_at: string;
}

export interface SessionInput {
  number: number;
  title: string;
  description?: string;
}

File: backend/internal/domain/session.go
package domain

import "time"

type Session struct {
	ID          int      `json:"id"`
	CourseID    int      `json:"course_id"`
	Number      int      `json:"number"`
	Title       string   `json:"title"`
	Description string   `json:"description"`
	TopicID     *int     `json:"topic_id,omitempty"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type SessionWithMaterials struct {
	Session
	Materials []Material
}

type SessionRepository interface {
	Create(session *Session) error
	GetByCourseID(courseID int) ([]Session, error)
	GetByID(id int) (*Session, error)
	Update(session *Session) error
	Delete(id int) error
}
